File: 1B.py
t = int(input())

for j in range(t):
    line = input().split()
    P = int(line[0])
    Q = int(line[1])

    X = [0]*(Q+1)
    Y = [0]*(Q+1)

    for i in range(P):
        line1 = input().split()
        if(line1[2] == 'N'):
            for i in range(int(line1[1])+1,Q+1):
                Y[i] += 1
        elif(line1[2] == 'S'):
            for i in range(0,int(line1[1])):
                Y[i] += 1

        elif(line1[2] == 'W'):
            for i in range(0,int(line1[0])):
                X[i] += 1

        else:
            for i in range(int(line1[0])+1,Q+1):
                X[i] += 1

    ans_x = 0
    ans_y = 0
    val_x = 0
    val_y = 0
    for i in range(Q+1):
        if(X[i] > val_x):
            val_x = X[i]
            ans_x = i

        if(Y[i] > val_y):
            val_y = Y[i]
            ans_y = i

    print(("Case #%d: %d %d") % (j+1,ans_x,ans_y))

File: 1B.cpp
#include<bits/stdc++.h>
#include <math.h>
#define MAX 2005
using namespace std;

int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int n = 8;
	int count = 0;

	for(int i=n; i>0; i/=2)
	{
		for(int j=0; j<i; j++)
			count += 1;
	}
	// x = x/2;
	// int i = 0;
	// for(i = 1; x>0; x = x/2)
	// 	i += 1;

	cout<<count;

	
	// int t;
	// cin>>t;
	// int T = t;
	// while(t--)
	// {   int P,Q,x,y;
	// 	char d;
	// 	int ans_x = 0, val_x = 0;
	// 	int ans_y = 0, val_y = 0;
    //     cin>> P >> Q;

	// 	int X[Q+1] = {0};
	// 	int Y[Q+1] = {0};

	// 	while(P--)
	// 	{
	// 		cin>> x >> y >> d;
	// 		if(d == 'N')
	// 		{
	// 			for(int i=y+1; i<=Q; i++)
	// 				Y[i]++;
	// 		}
	// 		else if(d == 'S')
	// 		{
	// 			for(int i=0; i<y; i++)
	// 				Y[i]++;
	// 		}
	// 		if(d == 'E')
	// 		{
	// 			for(int i=x+1; i<=Q; i++)
	// 				X[i]++;
	// 		}
	// 		if(d == 'W')
	// 		{
	// 			for(int i=0; i<x; i++)
	// 				X[i]++;
	// 		}
	// 	}

	// 	for(int i=0; i<=Q; i++)
	// 	{
	// 		if(X[i] > val_x)
	// 		{	val_x = X[i];
	// 			ans_x = i;
	// 		}

	// 		if(Y[i] > val_y)
	// 		{
	// 			val_y = Y[i];
	// 			ans_y = i;
	// 		}
	// 	}

	// 	cout<<"Case #"<<T-t<<": "<<ans_x<<" "<<ans_y<<endl;
    // }

	return(0);
}
